### PR TITLE
Add tcpKeepAliveListener to clean up connections where the client has disappeared

### PR DESCRIPTION
**Pull request by F21**
As mentioned in #67, it is good practice to clean up connections when the client has disappeared. The `tcpKeepAliveListener` is lifted from `net/http`: https://github.com/golang/go/blob/255e206b2bae9e7632043e08cf8cc0c7ce445c31/src/net/http/server.go#L2520

Unfortunately, `net/http` only applies the `tcpKeepAliveListener` to connections if you use `http`'s `ListenAndServe()` and `ListenAndServeTLS()` methods.

This PR brings that capability to graceful. There doesn't seem to be an easy way to automatically test this, hence the lack of tests in the PR and the original commit on `net/http`: https://github.com/golang/go/commit/d6bce32a3607222075734bf4363ca3fea02ea1e5

### CODE REVIEW

File: graceful.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ type Server struct {
 	// Limit the number of outstanding requests
 	ListenLimit int
 
+	// TCPKeepAlive sets the TCP keep-alice timeouts on accepted
```
Comment (tylerb) [inline graceful.go]: `alice`->`alive`

**Comment by tylerb** [comment]
Fix the typo and it LGTM. Happy to merge. Thanks again!

**Comment by F21** [comment]
@tylerb Thanks for catching that! It should be fixed! :smile:

**Comment by tylerb** [comment]
Btw I have added you as a collaborator. Thanks for your great work on these issues so far. If you feel comfortable merging your PR, please do so 😄

**Comment by F21** [comment]
Done :smile:

**Comment by tylerb** [comment]
One last thing. Can you also push a new tag for gopkg.in?

**Comment by F21** [comment]
Done :smile:

**Comment by tylerb** [comment]
Thanks!!

**Pull request merged**